mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp

```diff
@@ -1977,9 +1977,10 @@ LogicalResult OrderedRegionOp::verify() {
   if (getSimd())
     return failure();
 
-  if (auto container = (*this)->getParentOfType<WsloopOp>()) {
-    if (!container.getOrderedValAttr() ||
-        container.getOrderedValAttr().getInt() != 0)
+  if (auto loopOp = dyn_cast<LoopNestOp>((*this)->getParentOp())) {
```

Comment:
Could there be any case where there is another operation in between the OrderedRegionOp and the LoopNestOp? If not, maybe we should check the parent explicitly

Reply:
I wasn't too sure, so to be conservative I just replicated the previous behavior. However, looking at the OpenMP examples documents, I can see this is allowed:
```c
#pragma omp for ordered
for (i=0; i<n; i++) {
  if (i <= 10) {
    #pragma omp ordered
    work(i);
  }
  if (i > 10) {
    #pragma omp ordered
    work(i+1);
  }
}
```
So my guess is that it's possible that `omp.ordered.region` appears indirectly nested inside of an `omp.loop_nest`.

Reply:
I just made some changes to improve a bit the verification of ordered operations, as well as updating the tests. Let me know if this is a reasonable solution.

Reply:
Small ping @tblah. This is now the only thing left to be approved so that I can update and merge the whole PR stack. Thank you for your time!